distsql/select_result.go

```diff
@@ -196,6 +199,11 @@ func (r *selectResult) fetchResp(ctx context.Context) error {
 		if err != nil {
 			return errors.Trace(err)
 		}
+		r.mu.Lock()
```

Comment:
guess it's better to extract the for-body as a function and defer `r.mu.Unlock()` (if still use `sync.Mutex`).

Reply:
Done.